src/fit.jl

```diff
@@ -44,38 +48,44 @@ function fit!(
         LGBM_DatasetSetField(train_ds, "init_score", init_score)
     end
 
+    test_dss = []
+
+    for test_entry in test
+        test_ds = LGBM_DatasetCreateFromMat(test_entry[1], ds_parameters, train_ds, is_row_major)
+        LGBM_DatasetSetField(test_ds, "label", test_entry[2])
+        push!(test_dss, test_ds)
+    end
+
+    return fit!(estimator, train_ds, test_dss..., verbosity=verbosity)
+end
+
+
+# Pass Dataset class directly to speed up data loading
```

Comment:
specifically if the user has already constructed the dataset themselves and its part of an iterative process othwerwise it makes no difference